backend/capellacollab/extensions/modelsources/git/routes.py

```diff
@@ -161,25 +161,28 @@ def get_revisions(
         )
 
     url = git_model.path
+    username = git_model.username or ""
+    password = git_model.password or ""
+
     log.debug(
         "Fetch revisions of git-model '%s' with url '%s'", git_model.name, url
     )
 
-    git_env = os.environ.copy()
-    git_env["GIT_USERNAME"] = git_model.username or ""
-    git_env["GIT_PASSWORD"] = git_model.password or ""
-    for ref in ls_remote(url, git_env):
-        (_, ref) = ref.split("\t")
-        if "^" in ref:
-            continue
-        if ref.startswith("refs/heads/"):
-            remote_refs["branches"].append(ref[len("refs/heads/") :])
-        elif ref.startswith("refs/tags/"):
-            remote_refs["tags"].append(ref[len("refs/tags/") :])
-
+    remote_refs = get_remote_refs(url, username, password)
     remote_refs["default"] = git_model.revision
 
-    log.debug("Determined branches: %s", remote_refs["branches"])
-    log.debug("Determined tags: %s", remote_refs["tags"])
     log.debug("Determined default branch: %s", remote_refs["default"])
+
     return remote_refs
+
+
+@router.post("/revisions", response_model=GetRevisionsModel)
```

Comment:
From a semantic point of view, it would make the most sense to use a `GET` here. However, the problem is that the route gets the password in the body of the request and the body of a `GET` should have no meaning. Therefore it is best to use a `POST`, but in the future we might want to consider using `QUERY` https://www.ietf.org/archive/id/draft-ietf-httpbis-safe-method-w-body-02.html (if it is then supported in fast API).